components/search-filters/filter.tsx

```diff
@@ -1,74 +1,83 @@
-import { PropsWithChildren } from 'react';
+import { MouseEvent, ReactElement, useState } from 'react';
 import ButtonLink from '#components-ui/button';
 import {
   buildSearchQuery,
   IParams,
   ISearchFilter,
 } from '#models/search-filter-params';
-import { randomId } from '#utils/helpers';
+import { useOutsideClick } from 'frontend/src/hooks';
```

Comment:
`frontend`actually refers to the non react code. We want to empty the folder until it is eventually empty and we remove all the script toolchain

Maybe move hooks to another folder. either `./hooks` or `./utils.hooks`